.github/workflows/main.yml

```diff
@@ -40,9 +40,15 @@ jobs:
         if: matrix.os == 'macos-latest'
       - name: Install wasmtime for tests
         run: curl -f -L --retry 5 https://wasmtime.dev/install.sh | bash -s -- --version v2.0.2
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v3
         with:
-          submodules: true
+          fetch-depth: 0
+      # We can't use `--depth 1` here sadly because the GNU config
+      # submodule is not pinned to a particular tag/branch. Please
+      # bump depth (or even better, the submodule), in case of "error:
+      # Server does not allow request for unadvertised object" in the
+      # future.
```

Comment:
I guess if/when https://github.com/actions/checkout/issues/367 is fixed we could possibly revert this?

Reply:
Yes, by then we can use a single `actions/checkout` step to handle submodules as well.